lib/state/state.go

```diff
@@ -109,6 +110,14 @@ func LogDir(baseDir string, suffixes ...string) string {
 	return filepath.Join(append(elems, suffixes...)...)
 }
 
+// GravityInstallDir returns the location of the temporary state directory for
+// the install/join operation.
+// elems are appended to resulting path
+func GravityInstallDir(elems ...string) (path string) {
```

Comment:
Like I said in PR description, this is not 100% decided but I like a temporary state directory a lot more than anything in `/usr/share`. It is not without problems and I might end up using working directory instead or some combination (i.e. storing binary into working directory upon CURL download instead of `/usr/bin` to fix the current deficiency).

Reply:
The inconvenience with this I think is that you have to always run `gravity` from the same working dir. I.e. you may have installer in `~/tmp/xxx/yyy/zzz/installer` directory and always have to cd into it to execute any command. This is not a **huge** deal but I think it may be a bit confusing/frustrating from UX point of view. As a user, I would expect that once you've started install/join, you should be able to invoke it from anywhere on the node.

Is there a problem with picking stable sub-directory inside our "ephemeral" dir for example?

Another idea is to write "install/join working dir" location to our state file (/etc/.gravity I think it's called? same where state dir is stored) - gravity CLI commands read it first and can determine work dir this way.